M2/Macaulay2/packages/Permutations.m2

```diff
@@ -0,0 +1,383 @@
+newPackage(
+    "Permutations",
+    AuxiliaryFiles => true,
+    Version => "1.0", 
+    Date => "August 7, 2024",
+    Keywords => {"Combinatorics"},
+    Authors => {
+        {Name => "Sean Grate", 
+         Email => "sean.grate@auburn.edu", 
+         HomePage => "https://seangrate.com/"}
+    },
+    Headline => "functions for working with permutations"
+)
+
+export {
+    -- types
+    "Permutation",
+    -- methods
+    "permutation",
+    "toMatrix",
+    "cycleDecomposition",
+    "cycleType",
+    "ascents",
+    "descents",
+    "ascendingRuns",
+    "descendingRuns",
+    "exceedances",
+    "saliances",
+    "records",
+    "avoidsPattern",
+    "avoidsPatterns",
+    "isVexillary",
+    "isCartwrightSturmfels",
+    "isCDG",
+    "foataBijection",
+    "ord",
+    "sign",
+    "isEven",
+    "isOdd",
+    "isDerangement",
+    "fixedPoints",
+    "inversions",
+    -- symbols
+    "Weak"
+}
+
+-----------------------------------------------------------------------------
+-- **CODE** --
+-----------------------------------------------------------------------------
+------------------------------------
+-- Local utilities
+------------------------------------
+to1Index := w -> (w / (i -> i+1))
+to0Index := w -> (w / (i -> i-1))
```

Comment:
This is fine, but I'm curious if there was a problem with my suggestion about prepending 0 to switch to 0-index: https://github.com/Macaulay2/M2/issues/2581#issuecomment-2148157938

Reply:
No, those helper functions seemed like the most obvious thing to do at the time I had written the bulk of this code (December 2023).